Sources/SymDiffUtils/Utils.cs

```diff
@@ -993,67 +990,89 @@ public static string Make(List<Duple<string, int>> mTrace, List<Declaration> con
         {
-            SymDiffVC.collector = null;
+            collector = null;
         }
 
-        public static ProverInterface.Outcome MyBeginCheck(string descriptiveName, VCExpr vc, ProverInterface.ErrorHandler handler)
-        {
-            SymDiffVC.proverInterface.Push();
-            SymDiffVC.proverInterface.Assert(vc, true);
-            SymDiffVC.proverInterface.Check();
-            var outcome = SymDiffVC.proverInterface.CheckOutcomeCore(SymDiffVC.handler);
-            SymDiffVC.proverInterface.Pop();
-            return outcome;
-        }
+        // public static SolverOutcome MyBeginCheck(string descriptiveName, VCExpr vc, ProverInterface.ErrorHandler handler)
+        // {
+        //     SymDiffVC.proverInterface.Push();
+        //     SymDiffVC.proverInterface.Assert(vc, true);
+        //     SymDiffVC.proverInterface.Check();
+        //     var outcome = SymDiffVC.proverInterface.CheckOutcomeCore(SymDiffVC.handler, CancellationToken.None).Result;
+        //     SymDiffVC.proverInterface.Pop();
+        //     return outcome;
+        // }
         public static VCExpr GenerateVC(Program prog, Implementation impl)
         {
-            SymDiffVC.vcgen.ConvertCFG2DAG(impl);
+            SymDiffVC.vcgen.ConvertCFG2DAG(new ImplementationRun(impl, Console.Out));
             ModelViewInfo mvInfo;
-            var /*TransferCmd->ReturnCmd*/ gotoCmdOrigins = SymDiffVC.vcgen.PassifyImpl(impl, out mvInfo);
-
+            var /*TransferCmd->ReturnCmd*/ gotoCmdOrigins =
+                SymDiffVC.vcgen.PassifyImpl(new ImplementationRun(impl, Console.Out), out mvInfo);
+        
             var exprGen = SymDiffVC.proverInterface.Context.ExprGen;
             //VCExpr controlFlowVariableExpr = null; 
-            VCExpr controlFlowVariableExpr = /*CommandLineOptions.Clo.UseLabels ? null :*/ SymDiffVC.exprGen.Integer(BigNum.ZERO);
-
-
-            Dictionary<int, Absy> label2absy;
-            var vc = SymDiffVC.vcgen.GenerateVC(impl, controlFlowVariableExpr, out label2absy, SymDiffVC.proverInterface.Context);
+            VCExpr controlFlowVariableExpr = /*BoogieUtils.BoogieOptions.UseLabels ? null :*/ SymDiffVC.exprGen.Integer(BigNum.ZERO);
+        
+        
+            var absyIds = new ControlFlowIdMap<Absy>();
+            var vc = SymDiffVC.vcgen.GenerateVC(impl, controlFlowVariableExpr, absyIds, SymDiffVC.proverInterface.Context);
             VCExpr controlFlowFunctionAppl = SymDiffVC.exprGen.ControlFlowFunctionApplication(SymDiffVC.exprGen.Integer(BigNum.ZERO), SymDiffVC.exprGen.Integer(BigNum.ZERO));
             VCExpr eqExpr = SymDiffVC.exprGen.Eq(controlFlowFunctionAppl, SymDiffVC.exprGen.Integer(BigNum.FromInt(impl.Blocks[0].UniqueId)));
             vc = SymDiffVC.exprGen.Implies(eqExpr, vc);
-
-            SymDiffVC.handler = new VCGen.ErrorReporter(gotoCmdOrigins, label2absy, impl.Blocks, new Dictionary<Cmd, List<object>>(), SymDiffVC.collector, mvInfo, SymDiffVC.proverInterface.Context, prog);
+        
```

Comment:
You could probably remove this commented-out code.

Reply:
Done. I also uncommented and updated `MyBeginCheck` methods (defined identically in three places, wraps the check with a prover push & pop). These were and still are unused, with a [related comment](https://github.com/boogie-org/symdiff/pull/6/commits/91b5b15572deba17d2b21432d5cb8598abf34e35#diff-e14170ecdb9d3bf11ac11cf95bd2aeb6c37f77c79419a0e947e1be7caca6f769R1031):
```
//Use MyBeginCheck instead of BeginCheck as it is inconsistent with CheckAssumptions's Push/Pop of declarations
```
As far as I can tell `CheckAssumptions` is called only from rootcause and Dependency projects when extracting unsat cores, and not at all when using standard SymDiff.